lib/Dialect/TTNN/Analysis/L1InterleavedPolicy.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Dialect/TTNN/Analysis/L1InterleavedPolicy.h"
+#include "ttmlir/Dialect/TT/IR/TTOpsTypes.h"
+#include "ttmlir/Scheduler/Scheduler.h"
+
+namespace mlir::tt::ttnn {
+
+void L1InterleavedPolicy::run(
+    const std::unordered_set<Edge> &overrideReshardEdges) {
+  rootOp->walk([&](func::FuncOp func) {
+    mlir::tt::scheduler::Scheduler scheduler(&func);
+    llvm::SmallVector<mlir::Operation *> scheduleableOps;
+    Operation *currentOp = nullptr;
+    llvm::DenseMap<Operation *, tt::LayoutAttr> selectedOpLayout;
+
+    // TODO(fbajraktari):
+    // This is V0 implementation of L1 interleaved policy. In the current
+    // implementation we have a single L1ChainCofig per FuncOp. This implies
+    // that in case of DRAM spil we will have a disconnected chain of L1 ops.
+    // This will be fixed in V1.
```

Comment:
I don't like this proposed staging, disconnected chain is not a chain and this policy is just random then.

Reply:
This has already been fixed in V1 version of this policy (#1132).